lib/lte_link_control/lte_lc.c

```diff
@@ -151,32 +182,328 @@ static const char cgauth[] = "AT+CGAUTH="CONFIG_LTE_PDN_AUTH;
 static const char legacy_pco[] = "AT%XEPCO=0";
 #endif
 
-void at_handler(void *context, const char *response)
+enum lte_lc_notif_type {
+	LTE_LC_NOTIF_CEREG,
+	LTE_LC_NOTIF_CSCON,
+	LTE_LC_NOTIF_CEDRXP,
+
+	LTE_LC_NOTIF_COUNT,
+};
+
+static const char *const at_notifs[] = {
+	[LTE_LC_NOTIF_CEREG] = "+CEREG",
+	[LTE_LC_NOTIF_CSCON] = "+CSCON",
+	[LTE_LC_NOTIF_CEDRXP] = "+CEDRXP",
+};
+
+BUILD_ASSERT(ARRAY_SIZE(at_notifs) == LTE_LC_NOTIF_COUNT);
+
+static bool is_relevant_notif(const char *notif, enum lte_lc_notif_type *type)
```

Comment:
All modules interested in AT events or notifications need to write a function like this one. We could maybe add a helper function in the parser module to check the type of event/notification. cc @evenl 
It is too expensive to parse each notification or event in each subscriber module, so we really need to check its type before parsing it.

Reply:
Yes, that's absolutely woth looking more into